apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,48 @@
 'use strict';
+    var $block = null;
+    var $oldstack = null;
+$(document).ready(function () {
+    $('[data-stack]').click(function () {
+        var $this = $(this);
+        if ($block === null) {
+            $oldstack = $this;
+            $block = $oldstack.children().last().detach();
+        } 
+        else {
+            var $lastblock = $this.children().last().data("block");
+            if ($block.data("block") < $lastblock || !$lastblock) {
+                $this.append($block);
+                $block = null;
+                 checkForWin($this);
+            } else {
+                alert("please play a legal play");
```

Comment:
Good job throwing a user error here. For your portfolio, I recommend injecting your message directly in your HTML, perhaps the `#announce-winner` div would do.